File: marketplace-bakend/src/main/java/com/troyecto/marketplace/services/StoreService.java
package com.troyecto.marketplace.services;
import com.troyecto.marketplace.dtos.store.StoreRequest;
import com.troyecto.marketplace.dtos.store.StoreResponse;
import java.util.List;

public interface StoreService {
    StoreResponse RegisterNewStore(StoreRequest storeRequest);
    StoreResponse getStoreById(Long id);
    StoreResponse UpdateStore(Long id, StoreRequest storeRequest);
    String DeleteStore(Long id);
    List<StoreResponse> getStores();
}

File: marketplace_frontend/src/services/StoreService.js
import axios from "axios";

const API_URL="http://localhost:8080/api/stores";

export const getStores=()=> axios.get(API_URL);
export const getStoreById=(id)=> axios.get(`${API_URL}/${id}`);
export const RegisterNewStore=(store) => axios.post(API_URL,store);
export const DeleteStore=(id) => axios.delete(`${API_URL}/${id}`);
export const UpdateStore=(id,store) => axios.put(`${API_URL}/${id}`,store);
